src/components/tx/security/shared/TxSecurityContext.tsx

```diff
@@ -34,7 +34,12 @@ export const TxSecurityContext = createContext<{
 
 export const TxSecurityProvider = ({ children }: { children: JSX.Element }) => {
   const { safeTx, safeMessage } = useContext(SafeTxContext)
-  const [redefineResponse, redefineError, redefineLoading] = useRedefine(safeTx ?? safeMessage)
+  const data = useMemo(
```

Comment:
Should we move overwriting the `nonce` into the `useMemo` that creates `jsonData`?
Reason being that while Redefine does not care about the nonce - other providers probably do (as nonce 0 makes txs revert). Then the overwriting happens in `useRedefine` which is handling all redefine specific stuff.